File: java/Problem1040.java
import java.util.Scanner;

public class Problem1040 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double n1 = sc.nextDouble();
        double n2 = sc.nextDouble();
        double n3 = sc.nextDouble();
        double n4 = sc.nextDouble();

        // Cálculo da média com truncamento manual
        double media = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10.0;
        media = (int)(media * 10.0) / 10.0; // Trunca para 1 casa decimal

        System.out.printf("Media: %.1f\n", media);

        if (media >= 7.0) {
            System.out.println("Aluno aprovado.");
        } else if (media < 5.0) {
            System.out.println("Aluno reprovado.");
        } else {
            System.out.println("Aluno em exame.");
            double exame = sc.nextDouble();
            System.out.printf("Nota do exame: %.1f\n", exame);

            // Cálculo da média final com truncamento
            double mediaFinal = (media + exame) / 2.0;
            mediaFinal = (int)(mediaFinal * 10.0) / 10.0;

            System.out.println(mediaFinal >= 5.0 ? "Aluno aprovado." : "Aluno reprovado.");
            System.out.printf("Media final: %.1f\n", mediaFinal);
        }

        sc.close();
    }
}

File: python/problem1040.py
entrada = input().split();

n1 = float(entrada[0]);
n2 = float(entrada[1]);
n3 = float(entrada[2]);
n4 = float(entrada[3]);

#no final, tem que dar 10, entt os pesos são os máximos delas

#tenho que fazer o primeiro * o peso e dividir pela soma?

media = (n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1);

mediaFinal = media / 10;

print("Media: %.1f" % mediaFinal);

if mediaFinal >= 7:
	print("Aluno aprovado.");
elif mediaFinal < 5:
	print("Aluno reprovado.");
else:
	print("Aluno em exame.");
	valor = input();
	exame = float(valor);
	print("Nota do exame: %.1f" % exame);
	mediaFinal += exame;
	mediaFinal /= 2;
	if mediaFinal > 5:
		print("Aluno aprovado.");
		print("Media final: %.1f" % mediaFinal);
	else:
		print("Aluno reprovado.");
		print("Media final: %.1f" % mediaFinal);
